src/index.ts

```diff
@@ -227,11 +228,19 @@ function detectConcurrentRuns(context: WRunContext) {
       core.info(`Skip execution because a newer instance of the same workflow is running in ${newerRun.html_url}`);
       exitSuccess({ shouldSkip: true });
     }
-  } else if (context.concurrentSkipping === "same_content") {
+  } else if (context.concurrentSkipping === "same_content" || context.concurrentSkipping === "same_content_newer") {
     const concurrentDuplicate = concurrentRuns.find((run) => run.treeHash === context.currentRun.treeHash);
     if (concurrentDuplicate) {
-      core.info(`Skip execution because the exact same files are concurrently checked in ${concurrentDuplicate.html_url}`);
-      exitSuccess({ shouldSkip: true });
+      if (context.concurrentSkipping === "same_content") {
+        core.info(`Skip execution because the exact same files are concurrently checked in ${concurrentDuplicate.html_url}`);
+        exitSuccess({ shouldSkip: true });
+      } else if (context.concurrentSkipping === "same_content_newer") {
+        const concurrentIsOlder = concurrentRuns.find((run) => new Date(run.createdAt).getTime() < new Date(context.currentRun.createdAt).getTime());
```

Comment:
I am not so sure about this `concurrentIsOlder`. You are checking whether there exists an older run, but you are not checking whether this older run has the same content (the same treeHash)?